File: schemaorg-rs/src/member_program_tier.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///https://schema.org/additionalType
///https://schema.org/Text
///https://schema.org/URL
pub type MemberProgramTierAdditionalTypeFieldEnum = String;
///https://schema.org/MemberProgramTier
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct MemberProgramTier {
    #[serde(rename = "@context")]
    pub context: String,
    ///https://schema.org/alternateName
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///https://schema.org/image
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<MemberProgramTierImageFieldEnum>,
    ///https://schema.org/additionalType
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<MemberProgramTierAdditionalTypeFieldEnum>,
    ///https://schema.org/hasTierBenefit
    #[serde(rename = "hasTierBenefit")]
    #[serde_as(as = "OneOrMany<_>")]
    pub has_tier_benefit: Vec<TierBenefitEnumerationEnum>,
    ///https://schema.org/url
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///https://schema.org/identifier
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<MemberProgramTierIdentifierFieldEnum>,
    ///https://schema.org/membershipPointsEarned
    #[serde(rename = "membershipPointsEarned")]
    #[serde_as(as = "OneOrMany<_>")]
    pub membership_points_earned: Vec<MemberProgramTierMembershipPointsEarnedFieldEnum>,
    ///https://schema.org/disambiguatingDescription
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///https://schema.org/description
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<MemberProgramTierDescriptionFieldEnum>,
    ///https://schema.org/isTierOf
    #[serde(rename = "isTierOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub is_tier_of: Vec<MemberProgram>,
    ///https://schema.org/hasTierRequirement
    #[serde(rename = "hasTierRequirement")]
    #[serde_as(as = "OneOrMany<_>")]
    pub has_tier_requirement: Vec<MemberProgramTierHasTierRequirementFieldEnum>,
    ///https://schema.org/mainEntityOfPage
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<MemberProgramTierMainEntityOfPageFieldEnum>,
    ///https://schema.org/potentialAction
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///https://schema.org/sameAs
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///https://schema.org/name
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///https://schema.org/subjectOf
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<MemberProgramTierSubjectOfFieldEnum>,
}

File: schemaorg-ts/src/member_program_tier.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type MemberProgramTierAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/MemberProgramTier>
 **/
export interface MemberProgramTier {
  "@context": string;
  "@type": "MemberProgramTier";

  /**
   * <https://schema.org/hasTierBenefit>
   **/
  hasTierBenefit: all.OneOrMany<all.TierBenefitEnumerationEnum>;

  /**
   * <https://schema.org/hasTierRequirement>
   **/
  hasTierRequirement: all.OneOrMany<all.MemberProgramTierHasTierRequirementFieldEnum>;

  /**
   * <https://schema.org/membershipPointsEarned>
   **/
  membershipPointsEarned: all.OneOrMany<all.MemberProgramTierMembershipPointsEarnedFieldEnum>;

  /**
   * <https://schema.org/isTierOf>
   **/
  isTierOf: all.OneOrMany<all.MemberProgram>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.MemberProgramTierMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.MemberProgramTierIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.MemberProgramTierDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.MemberProgramTierSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.MemberProgramTierAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.MemberProgramTierImageFieldEnum>;
}
